### DISCUSSION TITLE
Is it possible to switch between themes using the TMUX_POWERLINE_THEME variable?

### DISCUSSION THREAD

**Discussion by Druva-D** [erikw/tmux-powerline#280] [Q&A] [answered]
I have multiple theme files in my themes directory, how do i change between themes without going to my powerlinerc and modifying the export? I tried exporting TMUX_POWERLINE_THEME from a bash script but that doesn’t work

**Comment by erikw** [comment]
Hey,

tmux-powerline will read the `~/.tmux-powerlinerc` every-time it refreshes (set by the tmux option `status-interval`). Thus if you change the `$TMUX_POWERLINE_THEME` in your `~/.tmux-powerlinerc`, you will immediately see that your theme has changed :)

**Comment by Druva-D** [comment]
Hi thanks for your response, here’s what i tried
~/.powerlinerc
```
    if [ -z '$COLORSCHEME']
    then
        export TMUX_POWERLINE_THEME=$COLORSCHEME
    else
        export TMUX_POWERLINE_THEME='tokyonight'
    fi
    echo "$COLORSCHEME 2" >> ~/.dotfiles/test.txt
```

in shell
```
~/.dotfiles rm test.txt
~/.dotfiles cat test.txt
 2
 ..
 2
~/.dotfiles tmux setenv COLORSCHEME 'onedark'
~/.dotfiles cat test.txt
 2
 ..
 2
~/.dotfiles export COLORSCHEME='tokyonight'
~/.dotfiles cat test.txt
 2
 ..
 2
```
How do I set the colorscheme variable?

**Comment by erikw** [comment, marked as answer]
As tmux-powerline will source you `config.sh` each time it renders, this is where we need to look. 

Usually the `~/.tmux-powerlinerc` looks like:
```bash
[...]
# The theme to use.
export TMUX_POWERLINE_THEME="default"
[...]
```

But as this file is sourced like shell script, you can add your own logic like setting the theme variable only if not already set like

```bash
# The theme to use. Don't override if already set in the environment.
if [ -z ${TMUX_POWERLINE_THEME+x} ]; then
        export TMUX_POWERLINE_THEME="default"
fi
```

Then tmux-powerline would use the value of `$TMUX_POWERLINE_THEME` from the evironment, not from the config, if set.

Then the question is how you would set this envvar? Setting it in a shell inside tmux will not have any effect, because that shell is unrelated to the shell that tmux will spawn when executing an update to render the left and right statuses. Maybe the shell that tmux spawns will read your shells initialization files like `~/.profile, ~/.bashrc, ~/.zshrc` or so . You need to experiment :)

However this does not sound very useful to me. If your goal is to be able to change the theme by typing

```console
$ export TMUX_POWERLINE_THEME="some-theme"
```

in a shell inside tmux, I would say this is not possible.

However the only thing you need to do is to update the ` ~/.tmux-powerlinerc` file really. So you can make a small bash script like

`set-tmux-powerline-theme.sh`:
```
#!/usr/bin/env bash

theme="$1"
echo "$theme" > $HOME/.tmux-powerline-theme
```

and change your  `~/.tmux-powerlinerc` to get the value from this file like

```bash
if [ -e $HOME/.tmux-powerline-theme ]; then
        export TMUX_POWERLINE_THEME="$(cat $HOME/.tmux-powerline-theme)"
fi 
```


Now you can change the theme by just doing
```
$ set-tmux-powerline-theme.sh theme1
$ set-tmux-powerline-theme.sh default
```

(and make the theme script executable and add it to your PATH of course)

I hope this can help you to develop your own solution :)

**Reply by Druva-D** [reply]
Yeah this works perfectly, thanks for helping
